### ISSUE TITLE
Action fails for remote repository where branch ${GITHUB_REF} does not exist

### ISSUE THREAD

**Issue by haertale** [GuillaumeFalourd/git-commit-push#5]
I use the action in a workflow job, where the result (a value) of a previous job is committed to a file in a remote repository. The repository of the workflow is not related to the remote repository at all (eg. one is not a fork of the other etc.).

The workflow looks like this:

    name: Example workflow with remote repository

    jobs:

      <multiple previous jobs>

      update-remote-repos:

        name: Update remote repository
        needs: <previous job>
        runs-on: ubuntu-latest

        steps:
          - name: Checkout remote repository
            uses: actions/checkout@v3
            with:
              repository: user/remote-repository
              token: ${{ secrets.PAT }}
              ref: some-branch

          - name: Update some file
            run: <modify ./some/file.dat>

          - Commit & push changes to remote repository
            uses: GuillaumeFalourd/git-commit-push@v1
            with:
              target_branch: some-branch
              file: ./some/file.dat
              access_token: ${{ secrets.PAT }}

The action will fail in case the workflow is triggered by a branch (eg. `main`) that does not exist in the remote repository - in the example above the job operating on the remote repository is working on a branch called `some-branch`, and let's assume that's the only branch of that repository.

Specifically the git fetch command will fail
https://github.com/GuillaumeFalourd/git-commit-push/blob/80a3d078073d53237559e58dabaaba07693734f0/action.yml#L92

as the script assumes the branch that triggered the workflow run is the current branch of the remote repository:

https://github.com/GuillaumeFalourd/git-commit-push/blob/80a3d078073d53237559e58dabaaba07693734f0/action.yml#L58

In other words, the script does not respect the branch that was specified during checkout of a remote repository using `actions/checkout`.

**Comment by GuillaumeFalourd** [comment]
Thank you for opening the issue @haertale 👍🏼 

Did this issue also happen using the `@v1.3` release tag, informing the `remote_repository` input? 

I observed you used the `@v1` in the example you gave, and some fixes has been implemented regarding the remote repository branches since then.
